lib/jackalope/work_list.ex

```diff
@@ -42,6 +43,35 @@ defmodule Jackalope.WorkList do
     %__MODULE__{work_list | items: tl(work_list.items)}
   end
 
+  @spec pending(t(), reference()) :: t()
+  def pending(work_list, ref) do
+    item = hd(work_list.items)
+
+    %__MODULE__{
+      work_list
+      | items: tl(work_list.items),
+        pending: Map.put(work_list.pending, ref, item)
+    }
+  end
+
+  @spec reset_pending(Jackalope.WorkList.t()) :: Jackalope.WorkList.t()
+  def reset_pending(work_list) do
+    pending_items =
+      for(
+        {ref, item} when is_reference(ref) <- work_list.pending,
+        do: item
+      )
+
+    prepend(%__MODULE__{work_list | pending: {}}, pending_items)
```

Comment:
```suggestion
    prepend(%__MODULE__{work_list | pending: %{}}, pending_items)
```
Make pending a map instead of a tuple.